controllers/volumereplicationgroup_controller.go

```diff
@@ -462,27 +468,132 @@ func (v *VRGInstance) createVolumeReplicationResources() bool {
 
 		volRep := &volrep.VolumeReplication{}
 
-		err := v.reconciler.Get(v.ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				log.Info("Creating VolumeReplication resource for PersistentVolumeClaim")
+		err := v.reconciler.Get(v.ctx, pvcNamespacedName, volRep)
+		if err != nil && errors.IsNotFound(err) {
+			log.Info("Uploading PersistentVolume")
 
-				if err = v.createVolumeReplicationForPVC(pvc); err == nil {
-					continue
-				}
+			if err = v.uploadPV(pvc); err != nil {
+				log.Error(err, "failed to upload PV metadata")
+
+				requeue = true
+
+				// There's no point enabling PV data replication if
+				// replication of PV metadata to object store has failed.
+				break
 			}
 
-			log.Info("Requeuing due to failure in getting or creating VolumeReplication resource for PersistentVolumeClaim",
-				"errorValue", err)
+			log.Info("Creating VolumeReplication resource for PersistentVolumeClaim")
 
-			// requeue on failure to ensure any PVC not having a corresponding VR CR
-			requeue = true
+			if err = v.createVolumeReplicationForPVC(pvc); err == nil {
```

Comment:
I think you meant `if err = v.createVolumeReplicationForPVC(pvc); err != nil {` notice the err == nil

Reply:
The err == nil check is correct, but the log message I added is incorrect.